pkg/registry/servicecatalog/broker/strategy.go

```diff
@@ -117,3 +127,29 @@ func (brokerRESTStrategy) ValidateUpdate(ctx kapi.Context, new, old runtime.Obje
 
 	return scv.ValidateBrokerUpdate(newBroker, oldBroker)
 }
+
+func (brokerStatusRESTStrategy) PrepareForUpdate(ctx kapi.Context, new, old runtime.Object) {
+	newBroker, ok := new.(*sc.Broker)
+	if !ok {
+		glog.Warning("received a non-broker object to update to")
```

Comment:
it'd be a good idea to bail out here. `newBroker` will be nil if `!ok`. same with the below functions